src/components/SeatLayout/SeatLayout.tsx

```diff
@@ -0,0 +1,143 @@
+// This function adds a seat to selectedSeat[] on clicking it
+// If seat is already selected then the function will unselect it
+// const updateSelectedSeats = (newSeat: number) => {
+//     if (selectedSeats.includes(newSeat)) {
+//         setSelectedSeats((prev) =>
+//             prev.filter((selectedSeat) => selectedSeat != newSeat)
+//         );
+//     } else setSelectedSeats((prev) => [...prev, newSeat]);
+// };
+//
+// This is how the seatLayout component is to be called when required.
+// <SeatLayout
+//     layoutName={layoutNames.volvo25}
+//     seats={seats}
+//     selectedSeats={selectedSeats}
+//     updateSelectedSeats={updateSelectedSeats}
+// />
+
+const SeatLayout = ({
+    layoutName,
+    seats,
+    selectedSeats,
+    updateSelectedSeats,
+}: {
+    layoutName: string;
+    seats: ISeat[];
+    selectedSeats: number[];
+    updateSelectedSeats: (seat: number) => void;
+}) => {
+    const [deck, setDeck] = useState<{
+        lowerDeck: number[][];
+    }>({ lowerDeck: [] });
+    const { t } = useTranslation('seatLayout');
+    let seatIndex = 0;
+
+    useEffect(() => {
+        setDeck((prev) => ({
+            ...prev,
+            lowerDeck: deckLayoutProducer(layoutConfig[layoutName].lowerDeck),
+        }));
+    }, [layoutName]);
+
+    return (
+        <SeatLayoutWrapper>
+            <div className="driver-cabin">
+                <img
+                    src={steeringWheel}
+                    alt="steering icon"
+                    className="steering"
+                />
+                <div className="line"></div>
+            </div>
+            <div className="seats-container">
+                {deck.lowerDeck.map((row: number[], rowIndex: number) => {
+                    return (
+                        <ul className="seat-row" key={`seatRow${rowIndex}`}>
+                            {row.map((seat: number, index: number) => {
+                                if (seat === 1) {
+                                    const currentSeat = seats[seatIndex];
+                                    seatIndex++;
+                                    return currentSeat.status.toString() ===
+                                        ISeatStatus.AVAILABLE.toString() ? (
+                                        selectedSeats.includes(
+                                            currentSeat.seatNumber
+                                        ) ? (
+                                            <Tooltip
+                                                title={`${
+                                                    currentSeat.seatNumber
+                                                }: ${t('selected')}`}
+                                                arrow
+                                                key={`seat${rowIndex}_${index}`}
+                                            >
+                                                <li
+                                                    className={`seat selected`}
```

Comment:
`seat selected` ?? is it working ?

Reply:
seat and selected are two different css classes. This code worked as expected.